### DISCUSSION TITLE
Scope Bugs: repos limit / can't share scopes / UI breaks (solved)

### DISCUSSION THREAD

**Discussion by Avivhdr** [community/community#11714] [Other Feature Feedback, Questions, & Ideas]
1/3 **Scopes are being stored in local storage**. this is less than optimal because:
- They can not be reused by other org/team members. 
- Hard refresh deletes them.

2/3 **there is a limit to the number of repos that can be added to a scope**.
I guess that this is because you can't the input can't contain more than 500 characters (in the `advanced` tab).
This means that I need to split my scope into 2 separate scopes

3/3 **When adding more the 8 repos, the UI breaks**:
![image](https://user-images.githubusercontent.com/23265867/154649202-3491fee9-2210-4d77-a9ee-bf441c9d2900.png)

**Comment by colinwm** [comment]
I've fixed the UI issue, so now you should be able to scroll that box. Sorry about that! 

As for the other issues:

> Scopes are being stored in local storage

I agree wtih the limitations of localstorage being unfortunate, but we won't change that in the immediate term.

They CAN be shared with teammates. If you send somebody a custom scope URL and they open it in their browser, they'll automatically get the custom scope added to their cs.github.com

 > there is a limit to the number of repos that can be added to a scope

I'll look into what I can do on this. Ultimately we may also run into limits with URL length as well... but perhaps it can be made a bit longer than it currently is

**Reply by Avivhdr** [reply]
1. Scrollbar works great. Thanks!

2. That's great, I didn't know that.

3. I hope there will be no such limitation. but until then I think that it would be better, and less confusing if there would be some kind of validation while the user adds repos to the scope. 

What happens now is that users can add as many repos as they want and the complete string will be truncated without them knowing about it. For me, it took a while until I released it. 

You can try it: create a new scope --> go to `advance` tab --> paste a string with 20 repositories --> go to `simple` tab: you will see that the list contains only some of the repositories that existed on the string.
